### PR TITLE
refactor: Breaking Changes since v1.4: Logout redirect (GH-5666)

### LINKED ISSUE #5666: Breaking Changes since v1.4: Logout redirect
1. Replace all constructors (overloads) with this constructor:
```
constructor(
    protected auth: AuthService,
    protected cms: CmsService,
    protected routing: RoutingService,
    protected semanticPathService: SemanticPathService,
    protected protectedRoutes: ProtectedRoutesService
  ) {}
```

### PR DESCRIPTION
**Pull request by bgambocjaviniar**
closes GH-5666

not sure if we remove the optional flag from the constructor. But I removed it according to what was said in the issue detail. 

Moreover, I am not sure if we're supposed to remove the param and add it back to make it 100% needed in the schematics.

### CODE REVIEW

File: docs/migration/2_0.md [hunk 1]
```diff
@@ -149,6 +149,7 @@ Support for functions dropped:
 - `CartVoucherService` now requires new parameter `ActiveCartService`. This service needs to be provided for `CartVoucherService`.
 - `CartCouponComponent` no longer uses `CartService`. This service usage was replaced with corresponding methods from `ActiveCartService`. Also `CustomerCouponService` and `FeatureConfigService` are now required parameters. These services needs to be provided for `CartCouponComponent`.
 - `CartItemListComponent` no longer uses `CartService`. This service usage was replaced with corresponding methods from `ActiveCartService.` Also `SelectiveCartService` and `FeatureConfigService` are now required parameters. These services needs to be provided for `CartItemListComponent`.
+- `LogoutGuard` no longer uses `FeatureConfigService`. This service needs to be provided for `LogoutGuard`.
```
Comment (tobi-or-not-tobi) [inline docs/migration/2_0.md]: > This service needs to be provided for `LogoutGuard`.

I'm confused about this, as the `FeatureConfigService` should no longer be used. I'd make it:

"`LogoutGuard` no longer uses `FeatureConfigService` as it was used previously to determine the feature flag."

File: projects/schematics/src/migrations/2_0/constructor-deprecations/constructor-deprecation-data.ts [hunk 2]
```diff
@@ -21,6 +21,8 @@ import { CHECKOUT_SERVICE_MIGRATION } from './data/checkout.service.migration';
 import { CONFIGURABLE_ROUTES_SERVICE_MIGRATION } from './data/configurable-routes.service.migration';
 import { CURRENT_PRODUCT_SERVICE_MIGRATION } from './data/current-product-service.migration';
 import { EXTERNAL_ROUTES_SERVICE_MIGRATION } from './data/external-routes.service.migration';
+import { LOGIN_FORM_COMPONENT_MIGRATION } from './data/login-form.component.migration';
```
Comment (tobi-or-not-tobi) [inline projects/schematics/src/migrations/2_0/constructor-deprecations/constructor-deprecation-data.ts:27]: I don't understand the notion of the `LOGIN_FORM_COMPONENT_MIGRATION` and `REGISTER_COMPONENT_MIGRATIONS` in this file. Seems from another PR.

**Review by tobi-or-not-tobi — DISMISSED** [review]
Good stuff. I've added a small comment and pushed a small optimisation and some code documentation.

File: projects/schematics/src/migrations/2_0/constructor-deprecations/constructor-deprecation-data.ts (hunk 2 above)
Comment (bgambocjaviniar) [reply projects/schematics/src/migrations/2_0/constructor-deprecations/constructor-deprecation-data.ts:27]: Yup, as mentioned in the other PR, it got moved due to the new prettier as its the import

line 41 and line 46 (I don't use those in my PR also as it's from develop)

File: docs/migration/2_0.md (hunk 1 above)
Comment (bgambocjaviniar) [reply docs/migration/2_0.md]: Thanks! will fix as I didn't know what to say with it and went ahead to make it somehow similar to the others

**Comment by bgambocjaviniar** [comment]
> Good stuff. I've added a small comment and pushed a small optimisation and some code documentation.

Cool, nice work and thanks for the feedback and optimisation

**Review by psmul — APPROVED** [review]

**Pull request merged**